pkg/envoy/generator/cds/builder_test.go

```diff
@@ -572,3 +572,38 @@ func TestRemoveDups(t *testing.T) {
 		},
 	}, removeDups(orig))
 }
+
+func TestBuild(t *testing.T) {
+	testCases := []struct {
+		name                string
+		builder             *clusterBuilder
+		numExpectedClusters int
+		expectErr           bool
+	}{
+		{
+			name: "OpenTelemetry ExtensionService cluster",
+			builder: &clusterBuilder{
+				openTelemetryExtSvc: &configv1alpha2.ExtensionService{
+					Spec: configv1alpha2.ExtensionServiceSpec{
+						Host:     "otel-collector",
+						Port:     4317,
+						Protocol: "h2c",
+					},
+				},
+			},
+			numExpectedClusters: 1,
+			expectErr:           false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			a := tassert.New(t)
+
+			actual, err := tc.builder.Build()
+
+			a.Len(actual, tc.numExpectedClusters)
```

Comment:
I'd prefer a stronger test if possible; couldn't you loop through the array and look for the expected cluster name?

Reply:
Actually we don't need to unmarshal in this test. I'll update to check the name.